### PR TITLE
Perspective UI modal views

### PR DESCRIPTION
**Pull request by gordonbrander**

### CODE REVIEW

File: type/browser/browser.js [hunk 1]
```diff
@@ -13,6 +14,7 @@ import * as Assistant from "./assistant"
 
 export type Model = {
   version: Version,
+  mode: PerspectiveUI.Model,
```
Comment (Gozala) [inline type/browser/browser.js]: Browser module should not be aware of perspective UI at all perspective UI should embed browser instead IMO.

Comment (gordonbrander) [inline type/browser/browser.js]: Ok, sounds good, but I'm unclear on what that should look like.

File: src/browser/sidebar.js [hunk 2]
```diff
@@ -107,3 +108,13 @@ export const view = ({entries}, address) =>
       className: 'sidebar-toolbar'
     })
   ]);
+
```
Comment (Gozala) [inline src/browser/sidebar.js]: I do think that embedder of side bar should just wrap it in a container that is displayed or not based on the mode off the embedder. Dose that make sense ?

Comment (gordonbrander) [inline src/browser/sidebar.js]: @Gozala that's not going to work when we start animating between modes. Transitions are more complex than just hide/show. We need to be able to modify the style of the root view as it transitions.

**Pull request closed without merging**